src/charm.py

```diff
@@ -0,0 +1,581 @@
+            ],
+        )
+        network_attachment_definition_spec = {
+            "config": json.dumps(
+                {
+                    "cniVersion": "0.3.1",
+                    "type": "macvlan",
+                    "ipam": {"type": "static"},
+                    "capabilities": {"mac": True},
+                }
+            )
+        }
+        self._kubernetes_multus = KubernetesMultusCharmLib(
+            charm=self,
+            containers_requiring_net_admin_capability=[self._bessd_container_name],
+            network_attachment_definitions=[
+                NetworkAttachmentDefinition(
+                    metadata=ObjectMeta(name=ACCESS_NETWORK_ATTACHMENT_DEFINITION_NAME),
+                    spec=network_attachment_definition_spec,
+                ),
+                NetworkAttachmentDefinition(
+                    metadata=ObjectMeta(name=CORE_NETWORK_ATTACHMENT_DEFINITION_NAME),
+                    spec=network_attachment_definition_spec,
+                ),
+            ],
+            network_annotations=[
+                NetworkAnnotation(
+                    name=ACCESS_NETWORK_ATTACHMENT_DEFINITION_NAME,
+                    interface=ACCESS_INTERFACE_NAME,
+                    ips=[self._access_network_ip],
+                ),
+                NetworkAnnotation(
+                    name=CORE_NETWORK_ATTACHMENT_DEFINITION_NAME,
+                    interface=CORE_INTERFACE_NAME,
+                    ips=[self._core_network_ip],
+                ),
+            ],
+        )
+
+    def _write_bessd_config_file(
+        self,
+        upf_hostname: str,
+        upf_mode: str,
+        access_interface_name: str,
+        core_interface_name: str,
+        dnn: str,
+        pod_share_path: str,
+    ) -> None:
+        """Write the configuration file for the 5G UPF service.
+
+        Args:
+            upf_hostname: UPF hostname
+            upf_mode: UPF mode
+            access_interface_name: Access network interface name
+            core_interface_name: Core network interface name
+            dnn: Data Network Name (DNN)
+            pod_share_path: pod_share path
+        """
+        jinja2_environment = Environment(loader=FileSystemLoader("src/templates/"))
+        template = jinja2_environment.get_template(f"{CONFIG_FILE_NAME}.j2")
+        content = template.render(
+            upf_hostname=upf_hostname,
+            mode=upf_mode,
+            access_interface_name=access_interface_name,
+            core_interface_name=core_interface_name,
+            dnn=dnn,
+            pod_share_path=pod_share_path,
+        )
+        self._bessd_container.push(
+            path=f"{BESSD_CONTAINER_CONFIG_PATH}/{CONFIG_FILE_NAME}", source=content
+        )
+        logger.info(f"Pushed {CONFIG_FILE_NAME} config file")
+
+    @property
+    def _upf_hostname(self) -> str:
+        """Returns the UPF hostname.
+
+        Returns:
+            str: UPF Hostname
+        """
```

Comment:
I have seen a lot of docstrings like this in recent reviews, and I would like to propose that we get rid of them for private properties that are that simple. The name of the property already says everything we need to know.

I would keep some docstrings for anything public, or when the docstring provides more details than the method signature. This is not the case here.

@Gmerold @saltiyazan @sanchezfdezjavier I think this is more of a team standard than anything, and therefore would like your input also.

Reply:
I agree